rust/lance/src/dataset.rs

```diff
@@ -1200,6 +1201,54 @@ impl Dataset {
         self.base.child(DATA_DIR)
     }
 
+    pub(crate) fn data_file_dir(&self, data_file: &DataFile) -> Result<Path> {
```

Comment:
Have you tested with indexes? We also have hard-coded paths and relative path references in indexes, I think we will need similar handling there.